custom_navbutton_completion.js

```diff
@@ -0,0 +1,119 @@
+                    altstr = M.util.get_string('completebuttontext', 'block_navbuttons');
+                    titlestr = M.util.get_string('completebuttontext', 'block_navbuttons');
+                    args.state.set('value', 1);
+                    if(btntype == 'icon') {
+                        args.image.set('src', M.util.image_url('tickicon', 'block_navbuttons'));
+                        args.image.set('alt', altstr);
+                        args.image.set('title', titlestr);
+                    }
+                    args.submit.set('value', altstr);
+                }
+            }
+
+            args.ajax.remove();
+        };
+
+        var handle_failure = function (id, o, args) {
+            alert('An error occurred when attempting to mark your activity complete.\n\n(' + o.responseText + '.)'); //TODO: localize
+            args.ajax.remove();
+        };
+
+        var toggle = function (e) {
+            e.preventDefault();
+
+            var form = e.target;
+            var cmid = 0;
+            var completionstate = 0;
+            var state = null;
+            var submit = null;
+            var image = null;
+            var modulename = null;
+            var btntype = null;
+
+            var inputs = Y.Node.getDOMNode(form).getElementsByTagName('input');
+            for (var i = 0; i < inputs.length; i++) {
+                switch (inputs[i].name) {
+                    case 'id':
+                        cmid = inputs[i].value;
+                        break;
+                    case 'completionstate':
+                        completionstate = inputs[i].value;
+                        state = Y.one(inputs[i]);
+                        break;
+                    case 'modulename':
+                        modulename = Y.one(inputs[i]);
+                        break;
+                    case 'btntype':
+                        btntype = Y.one(inputs[i]);
+                        break;
+                }
+                if (inputs[i].type == 'submit') {
+                    submit = Y.one(inputs[i]);
+                }
+                if (inputs[i].type == 'image') {
+                    image = Y.one(inputs[i]);
+                }
+            }
+
+            // start spinning the ajax indicator
+            var ajax = Y.Node.create('<div class="ajaxworking" />');
+            form.append(ajax);
+
+            var cfg = {
+                method: "POST",
+                data: 'id=' + cmid + '&completionstate=' + completionstate + '&btntype=' + btntype + '&fromajax=1&sesskey=' + M.cfg.sesskey,
+                on: {
+                    success: handle_success,
+                    failure: handle_failure
+                },
+                arguments: {state: state, submit: submit, image: image, btntype: btntype, ajax: ajax, modulename: modulename}
+            };
+
+            Y.use('io-base', function (Y) {
+                Y.io(M.cfg.wwwroot + '/course/togglecompletion.php', cfg);
```

Comment:
Same as above, retained original functionality for now and without it being a YUI module, not wrapping it breaks the functionality.